ibc_public/utils_data.py

```diff
@@ -574,86 +576,191 @@ def make_surf_db(derivatives=DERIVATIVES, conditions=CONDITIONS,
                     subjects.append(img.split('/')[-5])
                     contrasts.append(contrast)
-                    tasks.append(task_name)
-                    sides.append(side)
+                    tasks.append(task)
                     modalities.append('bold')
-                    meshes.append(mesh)
+                    supports.append(support)
                     acquisitions.append(acquisition)
 
-    print(f"{len(imgs)} images found, {len(missing_images)} were missing")
+            else:
+                for side in ['lh', 'rh']:
+                    selected_imgs_filename = os.path.join(
+                        derivatives, subject, '*', dir_, 'stat_maps',
+                        '%s_%s%s' % (contrast, side, extension)
+                    )
+                    if acquisition in ['ap', 'pa']:
+                        selected_imgs_filename = os.path.join(
+                            derivatives, subject, '*', dir_, 'z_score_maps',
+                            '%s*%s%s' % (contrast, side, extension)
+                        )
+                    imgs_ = glob.glob(selected_imgs_filename)
+                    imgs_.sort()
+
+                    # Display warning when no image is found
+                    if len(imgs_) == 0:
+                        missing_images_per_subject[subject].append(
+                            [subject, contrast, task, side]
+                        )
+
+                    for img in imgs_:
+                        session = img.split('/')[-4]
+                        imgs.append(img)
+                        sessions.append(session)
+                        subjects.append(img.split('/')[-5])
+                        contrasts.append(contrast)
+                        tasks.append(task)
+                        sides.append(side)
+                        modalities.append('bold')
+                        supports.append(support)
+                        acquisitions.append(acquisition)
 
-    # create a dictionary with all the information
-    db_dict = dict(
-        path=imgs,
-        subject=subjects,
-        contrast=contrasts,
-        session=sessions,
-        task=tasks,
-        side=sides,
-        modality=modalities,
-        mesh=meshes,
-        acquisition=acquisitions
-    )
+    total_missing_maps = sum([
+        len(missing_images_per_subject[subject])
+        for subject in missing_images_per_subject.keys()
+    ])
+    print(f"{len(imgs)} images found, {total_missing_maps} were missing")
+
+    for subject in missing_images_per_subject.keys():
+        print(f"Missing images for subject {subject}:\t{len(missing_images_per_subject[subject])}")
+
+    if support in volumetric_supports:
+        # create a dictionary with all the information
+        db_dict = dict(
+            path=imgs,
+            subject=subjects,
+            contrast=contrasts,
+            session=sessions,
+            task=tasks,
+            modality=modalities,
+            support=supports,
```

Comment:
semantically, this concept of 'support' seems a bit ad hoc. Isn't there a proper BIDS equivalent. 'space' ?

Reply:
I think you are right, `space` feels closer to standard BIDS vocabulary (I found more on this page https://bids-specification.readthedocs.io/en/stable/appendices/coordinate-systems.html#image-based-coordinate-systems).
Let's change support to space!